### PR TITLE
Delete bundle path if using bundle install --force

### PR DESCRIPTION
**Pull request by karlo57**
#3715

EDIT: #3715 doesn't apply anymore. But a new bug was found regarding `bundle install --force` reinstalling bundler. This pull request is for that.

### CODE REVIEW

**Comment by indirect** [comment]
Well, at a minimum this needs tests. :) This approach seems like it will probably work, though.

**Comment by segiddins** [comment]
This seems way overly-aggressive to me? We should only be cleaning up after the gems we're re-installing

**Comment by indirect** [comment]
If it's a bundle install --path --force it seems fine? But if it's --system --force, yes, this destroys everything. :P

**Comment by karlo57** [comment]
I'll add tests and look into cleaning up only the gems we're installing. Thanks guys!

**Comment by karlo57** [comment]
@indirect @segiddins I updated this to clean up after the gems we're only installing. I'll add tests tomorrow.

File: spec/install/force_spec.rb [hunk 1]
```diff
@@ -2,19 +2,41 @@
...
+    end
+
+    # How do I test this
+    it "deletes installed gems" do
+      gemfile <<-G
+        source "file://#{gem_repo1}"
+        gem "rack"
+      G
       bundle "install"
+
```
Comment (indirect) [inline spec/install/force_spec.rb]: On this line, add a file into the directory where `rack` is installed. IIRC, that's something like `default_bundle_path("gems/rack-1.0.0/garbage").open{|f| f.write("blah blah blah") }`.

File: spec/install/force_spec.rb [hunk 2]
```diff
@@ -2,19 +2,41 @@
...
+      bundle "install --force"
+
```
Comment (indirect) [inline spec/install/force_spec.rb]: On this line, check to make sure the file you added no longer exists. For example, `expect(default_bundle_path("gems/rack-1.0.0/garbage")).to_not exist`.

**Comment by indirect** [comment]
See the inline comments for testing suggestions. :)

**Comment by karlo57** [comment]
I added the lines to the test. It works. Although now I'm seeing that this is a non-issue? I commented the  lines which uninstall the gems and the spec still passes. I also have verified that the garbage file is actually written so there's nothing wrong with the tests.

**Comment by indirect** [comment]
Well, I guess I was right about there being a bug, but only by accident! Heh. Thanks for writing these tests, they show the real bug!

```
Installing bundler 1.10.4

Bundler::GemNotFound: Could not find bundler-1.10.4.gem for installation
Installing rack 1.0.0
An error occurred while installing bundler (1.10.4), and Bundler cannot continue.
Make sure that `gem install bundler -v '1.10.4'` succeeds before bundling.
```

**Comment by indirect** [comment]
I wound up making these changes to show that problem:

```
Installing bundler 1.10.4

Bundler::GemNotFound: Could not find bundler-1.10.4.gem for installation
Installing rack 1.0.0
An error occurred while installing bundler (1.10.4), and Bundler cannot continue.
Make sure that `gem install bundler -v '1.10.4'` succeeds before bundling.
```

**Comment by indirect** [comment]
Looks like we don't need to actively uninstall, but we definitely need these tests, and we need to not force-install Bundler itself.

**Comment by karlo57** [comment]
Cool, I updated the code now. Please review. Thanks!

**Comment by indirect** [comment]
Merged as 13c445680f65293ee50f11718e8a86bfa87abd2e. Thanks!

**Pull request closed without merging**